core/controllers/ingest.py

```diff
@@ -78,6 +86,36 @@ def ingest(body: dict, excel_file: FileStorage) -> Response:
     return jsonify(success_payload)
 
 
+def process_validation_failures(validation_failures):
+    internal_failures = [failure for failure in validation_failures if isinstance(failure, InternalValidationFailure)]
```

Comment:
docs and type hints for this please

also, we'll need some unit tests for this function as well I think. We may be able to use some of the existing ones from when this was used as a handler function?

Best to just mock out the return value of `faulures_to_messages` etc